File: javaexample/src/main/java/io/chthonic/mythos/javaexample/ui/vus/RoVu.java
package io.chthonic.mythos.javaexample.ui.vus;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import io.chthonic.mythos.javaexample.R;
import io.chthonic.mythos.javaexample.databinding.FragmentRoBinding;
import io.chthonic.mythos.javaexample.ui.layouts.DahLayout;
import io.chthonic.mythos.mvp.Vu;

/**
 * Created by jhavatar on 3/22/2016.
 */
public class RoVu extends Vu<FragmentRoBinding> {

    private Runnable toggleDahListener = null;

    public RoVu(LayoutInflater layoutInflater,
                Activity activity,
                Fragment fragment,
                ViewGroup parentView) {

        super(layoutInflater, activity, fragment, parentView);
    }

    @Override
    public void onCreate() {
        Button toggleButton = this.getRootView().findViewById(R.id.button_toggle_dah);
        toggleButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Runnable listener = toggleDahListener;
                if (listener != null) {
                    listener.run();
                }
            }
        });
    }

    public Runnable getToggleDahListener() {
        return toggleDahListener;
    }

    public void setToggleDahListener(Runnable toggleDahListener) {
        this.toggleDahListener = toggleDahListener;
    }

    public void updateDahDisplay(boolean showDah) {
        ViewGroup rootLayout = (ViewGroup) getRootView();
        View dahLayout = getRootView().findViewById(R.id.dah_layout);
        if (showDah && (dahLayout == null)) {
            dahLayout = new DahLayout(getActivity(), getActivity().getResources().getString(R.string.ro_lifecycle_key));
            dahLayout.setId(R.id.dah_layout);
            rootLayout.addView(dahLayout);

        } else if (!showDah && (dahLayout != null)) {
            rootLayout.removeView(dahLayout);
        }
    }

    public void setText(String text) {
        ((TextView) getRootView().findViewById(R.id.ro_text)).setText(text);
    }

}

File: kotlinexample/src/main/kotlin/io/chthonic/mythos/kotlinexample/ui/vus/RoVu.kt
package io.chthonic.mythos.kotlinexample.ui.vus

import android.app.Activity
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import io.chthonic.mythos.kotlinexample.R
import io.chthonic.mythos.kotlinexample.ui.layouts.DahLayout
import io.chthonic.mythos.mvp.FragmentWrapper
import io.chthonic.mythos.mvp.Vu
import kotlinx.android.synthetic.main.fragment_ro.view.*

/**
 * Created by jhavatar on 3/22/2016.
 */
class RoVu (inflater: LayoutInflater,
            activity: Activity,
            fragmentWrapper: FragmentWrapper? = null,
            parentView: ViewGroup? = null) :
        Vu(inflater,
                activity = activity,
                fragmentWrapper = fragmentWrapper,
                parentView = parentView) {

    var toggleDahListener: (() -> Unit)? = null

    override fun onCreate() {
        super.onCreate()

        rootView.button_toggle_dah.setOnClickListener { v ->
            toggleDahListener?.invoke()
        }
    }

    override fun getRootViewLayoutId(): Int {
        return R.layout.fragment_ro
    }

    fun updateDahDisplay(show: Boolean) {
        val dah = rootView.dah_layout

        if (show && (dah == null)) {
            val roLayout = DahLayout(activity, activity.resources.getString(R.string.ro_lifecycle_key))
            roLayout.id = R.id.dah_layout
            (rootView as ViewGroup).addView(roLayout)

        } else if (!show && (dah != null)) {
            (rootView as ViewGroup).removeView(dah)
        }
    }

}